libr/magic/apprentice.c

```diff
@@ -276,8 +280,9 @@ struct mlist * file_apprentice(RMagic *ms, const char *fn, size_t fn_size, int a
 	char *p, *mfn;
 	int file_err, errs = -1;
 	struct mlist *mlist;
+	bool done = false;
 
-	init_file_tables ();
+	init_file_tables (&done);
```

Comment:
that will initialize the tables everytime you call file_apprentice. the current implementation initalizes those tables only ONCE per thread and execution, which is not the same.

You should move this initialization into the r_magic_new() and then you can remove the init_file_tables() call inside the file_apprentice.

i have pusehd this change which superseeds this pr https://github.com/radareorg/radare2/pull/20474